FCCEUD-24 describe searching for modules

emarcusRH opened the pull request.

adahms commented:
Thanks for the PR, @emarcusRH!

Wondering whether we need a more procedural approach to back up some of the finer points users might run into, such as being able to select multiple products, using and dismissing the product tiles after they've been added, and how free text works.

For example, the Satellite docs draw up some examples and describe fields to help provide some more context and cover common problems -

https://access.redhat.com/documentation/en-us/red_hat_satellite/6.6/html-single/administering_red_hat_satellite/index#chap-Red_Hat_Satellite-Administering_Red_Hat_Satellite-Searching_and_Bookmarking

robin-owen commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/ref_search_modules.adoc:
```diff
@@ -0,0 +1,20 @@
+* The module title
+* Publication state - published or not published
+* The date the draft was uploaded
+* Module type.
```
Periods are used for these two bullet points, but not for the others. We should be consistent one way or the other. :)

robin-owen commented:
By the way, I agree with @adahms about using a more procedure-oriented approach. At first, I thought this was inaccurately written as a reference module and should have been a few procedures.

emarcusRH commented:
> Thanks for the PR, @emarcusRH!
> 
> Wondering whether we need a more procedural approach to back up some of the finer points users might run into, such as being able to select multiple products, using and dismissing the product tiles after they've been added, and how free text works.
> 
> For example, the Satellite docs draw up some examples and describe fields to help provide some more context and cover common problems -
> 
> https://access.redhat.com/documentation/en-us/red_hat_satellite/6.6/html-single/administering_red_hat_satellite/index#chap-Red_Hat_Satellite-Administering_Red_Hat_Satellite-Searching_and_Bookmarking
@adahms , @benradey 
What is the full functionality of the search bar in PII ?
does it support all the boolean queries listed in the Satellite search chapter? 
Do we really need to spell out all the parameters, switches and options for a user?

emarcusRH commented:
@adahms @benradey @robin-owen 
- I changed the module/file type to **proc-**
- added a procedural approach
- updated some of the names according to latest UI instance that I can access 
([P2 QA](https://pantheon2-qa.int.us-east.aws.preprod.paas.redhat.com/pantheon/#/)
- I still am not 100% clear on the functionality of the last drop down menu (different properties)
- I also have not been able to see the Sort button have any effect on the results table

adahms commented:
Thanks for the updates, @emarcusRH!

In response to your question above, Pantheon does not support the operators mentioned in the Satellite guide - I am just providing that as a reference to how we can structure some of the details, either as tables or procedures.

As for the functionality itself, I call out some of the foibles users might encounter and not find intuitive at first glance in the JIRA. As long as we cover those, either as steps, a reference, or admonitions, I think that's enough for now.

I like the updates you've made to the module as well. On reviewing it, how about we structure this as an assembly with a few modules - searching for modules, with a procedure for free text search, filtering, and viewing?

benradey commented:
> > What is the full functionality of the search bar in PII ?
> > does it support all the boolean queries listed in the Satellite search chapter?

No, the current functionality of the search bar is a simple plain-text search of the "title" and "description" fields of documents in Pantheon.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-searching_modules.adoc:
```diff
@@ -0,0 +1,26 @@
+[id="search-modules"]
+
+= How to search Modules
```
@emarcusRH - How about making this into an assembly? You have the basis for several procedures, which should each be in their own module, and the introduction you have here sounds great as an introduction to a use case.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-searching_modules.adoc:
```diff
@@ -0,0 +1,26 @@
+.To perform a keyword or free text search:
+. In the search bar at top right, enter a free text phrase or keywords.
+. You can also add Boolean operators to the keywords to refine the search.
+. Press `Enter` or click the search icon  to initiate the search.
```
There's a double space here that can be collapsed.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-searching_modules.adoc:
```diff
@@ -0,0 +1,26 @@
+. Press `Enter` or click the search icon  to initiate the search.
+
+.To add filters to the search, use any combination of the following:
+* Select a Product name from the `Select a Product` drop down menu.
```
s/Product name/product name/
s/drop down menu/menu/

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-filters.adoc:
```diff
@@ -0,0 +1,8 @@
+[id='search-filters']
```
Needs '_{context}' in ID.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-keyword.adoc:
```diff
@@ -0,0 +1,6 @@
+[id='search-keyword']
```
Needs '_{context}' in ID.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,10 @@
+[id='view-results']
```
Needs '_{context}' in ID.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-filters.adoc:
```diff
@@ -0,0 +1,8 @@
+[id='search-filters']
+
+.To add filters to the search, use any combination of the following:
```
Should be '.Procedure'

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-keyword.adoc:
```diff
@@ -0,0 +1,6 @@
+[id='search-keyword']
+
+.To perform a keyword or free text search:
```
Should be '.Procedure'

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,10 @@
+[id='view-results']
+
+.Viewing search results
```
Should be '.Procedure'

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-filters.adoc:
```diff
@@ -0,0 +1,8 @@
+[id='search-filters']
+
```
Needs an introduction sentence.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,10 @@
+[id='view-results']
+
```
Needs an introduction sentence.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,10 @@
+[id='view-results']
+
+.Viewing search results
+The search results are presented in a table organized into columns with the following headings:
```
Needs an introduction sentence - what does this procedure help the user achieve?

benradey commented:
@emarcusRH I see that you removed "WIP" from the title of this PR - is it ready for tech review?

emarcusRH commented:
> WIP

yes, I think I'm ready to finalize thisw as well as FCCEUD-25

benradey commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-filters.adoc:
```diff
@@ -0,0 +1,14 @@
+* Select a Product name from the `Select a Product` menu.
+* Select a version number from the `Select a Version` menu.
+* Select a module type from the third menu (`All`, `Concept`, `Procedure`, or `Reference`).
+* Select a property from the fourth menu (`Uploaded date`, `Title`, `Updated date`, or `Module type`)
```
Probably worth noting that this is for sorting purpose only (it's not a filter like the other 3) and only applies once the search is invoked.

benradey commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-search-keyword.adoc:
```diff
@@ -0,0 +1,9 @@
+.Procedure
+
+. In the search bar at top right, enter a free text phrase or keywords.
+. You can add Boolean operators to the keywords to refine the search.
```
I don't think boolean operators are supported, unless I misunderstand what you mean here.

I searched "Assembly" on my system with 20 documents and got 1 result.

I searched for "-Assembly" and expected to get the other 19, but I got 0.

benradey commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,10 @@
+* `Published` - the publication status, `Published` or `Not published`
+* `Draft uploaded` - the date the draft was uploaded
+* `Module Type` - the module type, `Concept`, `Procedure`, or `Reference`
+* A sorting button allows you to quickly change the sorting order of search results from A-Z to Z-A.
```
Again, just a note that the sort button only applies to the next search, not the current set of results.

benradey commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/ref_command-line-interface.adoc:
```diff
@@ -0,0 +1,45 @@
+pantheon push -u <YOUR_USER> -p <YOUR_PASSWORD>
+----
+. Locate and preview your modules at -
+link:http://pantheon2.int.us-west.aws.prod.paas.redhat.com/pantheon/[]
```
I want to point out that our prod environment is locked down so that users cannot push modules directly using this method. They _must_ go through the UI and perform a git sync.

In the lower environments (dev, stage, etc) this is still possible.

Or, if some 3rd party community wanted to adopt a Pantheon instance, they could configure it to allow this.

So because of the variability, I'm not sure what makes sense to document. I'll leave that to your team to decide.

benradey commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/ref_command-line-interface.adoc:
```diff
@@ -0,0 +1,45 @@
+----
+. Change the default values for ‘server’ and ‘repository’ as follows:
++
+server: link:http://pantheon2.int.us-west.aws.prod.paas.redhat.com[]
```
Are we trying to avoid publishing internal Red Hat URLs?

adahms replied:
@benradey - yes.

@emarcusRH - can we change this to a more generic reference?

benradey commented:
Looks good to me! Thanks for the updates, @emarcusRH. We would still be waiting on a peer approval (perhaps from @adahms or @robin-owen) in order to merge.

robin-owen commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc-view-results.adoc:
```diff
@@ -0,0 +1,11 @@
+.Viewing search results
+The search results are presented in a table organized into columns with the following headings:
+
+* `Title` - the module title
+* `Published` - the publication status, `Published` or `Not published`
+* `Draft uploaded` - the date the draft was uploaded
+* `Module Type` - the module type, `Concept`, `Procedure`, or `Reference`
```
Similar to the filters in another file, only these four are actual column headings. The next two items in this list describe an actual procedure.

emarcusRH commented:
@redhataccess/eud-tech-review 
final peer review and updates completed - ready for merge!

benradey commented:
@emarcusRH Could I request that either @adahms or @robin-owen provide a github approval to formally complete the peer review step?

![image](https://user-images.githubusercontent.com/22244893/85859990-0840f280-b78c-11ea-82fc-b8566d3b2f1a.png)

benradey commented:
Gracias!
